pkg/tenancy/config.go

```diff
@@ -0,0 +1,133 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package tenancy
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+)
+
+// TenantLabelKey is a label key reserved for tenancy.
+const TenantLabelKey = "__tenant__"
+
+// AuthConfig defines configuration type for tenancy.
+type AuthConfig interface {
+	// AllowNonTenants returns true if tenancy is asked to accept write-requests from non-multi-tenants.
+	allowNonTenants() bool
```

Comment:
Comment and name don't match. Should this be exported?

Reply:
comment updated. I don't think exporting this is required.